### DISCUSSION TITLE
Backups to remote storage: Velero?

### DISCUSSION THREAD

**Discussion by ShaunMaher** [openebs/openebs#3643] [Q&A] [answered]
Hi.

I have had a need in the past few days to restore a PV from backup.  It didn't work.  No data was lost that couldn't be recovered by other means so it wasn't a huge deal.  I don't have any important data on Kubernetes yet so now is the time to learn from failures. :-)

I'm having trouble understanding how OpenEBS cStor and Velero achieve backups to remote storage.  I've searched the doco and I can't find anything that really explains "how" Velero + OpenEBS achieve their magic.

I have a BackupStorageLocation configured and I have scheduled a backup job to TODO to that location.  Files are created in the bucket, Velero says the backups have been successful.  I have the backup TTL set to 10 days.

I suspect that the files that are created by Velero in the remote bucket called, for example "cstor-pvc-18fb6caf-ce69-48b4-8ca1-b34c8702a79f-velero-daily-backup-20230620040854" (without the .pvc extension) might be ZFS data streams?  As-in, the result of a "zfs send" command.

If this is true, they seem quite small compared to the source PV size so I'm assuming they are incremental data streams, between ZFS snapshots.

What I can't seem to find is an initial "full" backup that the incrementals are based on.  I have a suspicion that the "full" backups occurred more than 10 days ago and have been deleted by Velero.  There doesn't seem to be a way to tell the OpenEBS Velero plugin that I would like a periodic full backup.

Am I way off the mark here with chained bad assumptions?

Can someone smarter than me please give me a dot point overview of how the PV backups of OpenEBS cStor volumes works with Velero.

Cheers.
Shaun.

**Comment by avishnu** [comment, marked as answer]
Thanks for being a user of OpenEBS !!
I'd like to suggest Mayastor as the alternative to CStor.
Mayastor is a replicated storage solution (just like CStor) but it's very fast as compared to CStor.
Mayastor is under active development and support, unlike CStor.

https://github.com/openebs/mayastor
https://mayastor.gitbook.io/introduction/

We have a roadmap item certify backup/restore of Mayastor storage volumes with Velero.
